Werewolf for Telegram/Werewolf Node/Werewolf.cs

```diff
@@ -4626,63 +4631,140 @@ private bool DoGameEnd(ITeam team)
-                                        deathmessage += Environment.NewLine + GetLocaleString("ThiefEnd", thief.GetName());
+                                        deathmessage += Environment.NewLine + GetLocaleString($"{thiefOrDG.PlayerRole}End", thiefOrDG.GetName());
+                                    }
+                                }
+                                //thief and doppelganger
+                                else if (alives.Any(x => x.PlayerRole == IRole.Thief) && alives.First(x => x.PlayerRole != IRole.Thief).PlayerRole == IRole.Doppelgänger)
+                                {
+                                    var thief = alives.FirstOrDefault(x => x.PlayerRole == IRole.Thief);
+                                    var doppelganger = alives.FirstOrDefault(x => x.PlayerRole == IRole.Doppelgänger);
+                                    
+                                    if (thief != null && doppelganger != null)
+                                    {
+                                        deathmessage = GetLocaleString("ThiefEnd", sorc.GetName()) + Environment.NewLine;
+                                        deathmessage += Environment.NewLine + GetLocaleString("DoppelgängerEnd", thiefOrDG.GetName());
                                     }
                                 }
                                 break;
 
-                            case 1: // Tanner or sorcerer or thief
-                                var lastone = alives.FirstOrDefault();
+                            case 1: // Tanner or sorcerer or thief or doppelgänger
+                                var lastone = alives.FirstOrDefault(x => x.PlayerRole == IRole.Sorcerer || x.PlayerRole == IRole.Thief || x.PlayerRole == IRole.Doppelgänger);
                                 if (lastone != null)
                                 {
                                     if (lastone.PlayerRole == IRole.Tanner)
                                     {
                                         KillPlayer(lastone, KillMthd.Suicide, killer: lastone, isNight: false);
-
-                                        deathmessage = GetLocaleString("TannerEnd", lastone.GetName());
                                     }
                                     else if (lastone.PlayerRole == IRole.Sorcerer)
                                     {
-                                        deathmessage = GetLocaleString("SorcererEnd", lastone.GetName());
                                         AddAchievement(lastone, AchievementsReworked.TimeToRetire);
                                     }
-                                    else if (lastone.PlayerRole == IRole.Thief)
-                                    {
-                                        deathmessage = GetLocaleString("ThiefEnd", lastone.GetName());
-                                    }
+                                    deathmessage = GetLocaleString($"{lastone.PlayerRole}End", lastone.GetName());
```

Comment:
I do like this change though :)

Reply:
i will fix all  asap thanks alot for noting and checking

Reply:
@Olgabrezel  $"{lastone.PlayerRole}End" for doppelganger this will be "DoppelgängerEnd" or "DoppelgangerEnd" ?